sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -75,31 +112,46 @@ export interface MessageHandlerOptions {
     maxMessageAutoRenewLockDurationInSeconds?: number;
 }
 
-export { MessagingError }
-
 // @public
-export interface OnError {
-    (error: MessagingError | Error): void;
+export interface MessageHandlers<ContextT> {
+    processError(err: Error): Promise<void>;
+    processMessage(message: ReceivedMessage, context: ContextT): Promise<void>;
 }
 
 // @public
-export interface OnMessage {
-    (message: ServiceBusMessage): Promise<void>;
-}
+export type MessageIterator<ContextT> = AsyncIterable<MessageAndContext<ContextT>>;
+
+export { MessagingError }
 
-// Warning: (ae-forgotten-export) The symbol "Client" needs to be exported by the entry point index.d.ts
-//
 // @public
-export class QueueClient implements Client {
+export interface NonSessionReceiver<LockModeT extends "peekLock" | "receiveAndDelete"> {
     close(): Promise<void>;
-    createReceiver(receiveMode: ReceiveMode): Receiver;
-    createReceiver(receiveMode: ReceiveMode, sessionOptions: SessionReceiverOptions): SessionReceiver;
-    createSender(): Sender;
-    readonly entityPath: string;
-    static getDeadLetterQueuePath(queueName: string): string;
-    readonly id: string;
-    peek(maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-    peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
+    diagnostics: {
+        peek(maxMessageCount?: number): Promise<ReceivedMessage[]>;
+        peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessage[]>;
+    };
+    // (undocumented)
+    entityPath: string;
+    // (undocumented)
+    entityType: "queue" | "subscription";
+    // (undocumented)
+    getDeadLetterPath(): string;
+    getMessageIterator(options?: GetMessageIteratorOptions): MessageIterator<ContextType<LockModeT>>;
+    // (undocumented)
+    isReceivingMessages(): boolean;
+    receiveBatch(maxMessages: number, maxWaitTimeInSeconds?: number, options?: ReceiveBatchOptions): Promise<{
+        messages: ReceivedMessage[];
+        context: ContextType<LockModeT>;
+    }>;
+    // (undocumented)
+    receiveDeferredMessage(sequenceNumber: Long, options?: OperationOptions): Promise<ServiceBusMessage | undefined>;
+    // (undocumented)
+    receiveDeferredMessages(sequenceNumbers: Long[], options?: OperationOptions): Promise<ServiceBusMessage[]>;
+    // (undocumented)
+    receiveMode: "peekLock" | "receiveAndDelete";
+    // (undocumented)
+    renewMessageLock(lockTokenOrMessage: string | ReceivedMessage): Promise<Date>;
```

Comment:
In .Net we have removed the` renewMessageLock` overload where we take `lockToken` as a parameter. Do we need the same here?

Reply:
Should fix, should only take the Message

Reply:
If we have to choose between the two, I would vote for lock token.
Users having access to message will have access to the lock token as well, but vice versa doesnt hold true

Reply:
What is the reason to have the renewMessageLock on the receiver, but the settlement methods on the context?

Reply:
@ramya-rao-a  - you wouldn't believe what I already have a TODO item for :)

Reply:
Moved to message under `renewLock()`